src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.cpp

```diff
@@ -36,38 +55,79 @@ void Dynamics::print(std::ostream& anOutputStream, bool displayDecorator) const
     displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void();
 }
 
-NumericalSolver::SystemOfEquationsWrapper Dynamics::GetDynamicalEquations(
-    const Array<Shared<Dynamics>>& aDynamicsArray, const Instant& anInstant
+NumericalSolver::SystemOfEquationsWrapper Dynamics::GetSystemsOfEquations(
+    const Array<Dynamics::Context>& aContextArray,
+    const Instant& anInstant,
+    const Shared<const Frame>& aFrame
 )
 {
     return std::bind(
         Dynamics::DynamicalEquations,
         std::placeholders::_1,
         std::placeholders::_2,
         std::placeholders::_3,
-        aDynamicsArray,
-        anInstant
+        aContextArray,
+        anInstant,
+        aFrame
     );
 }
 
 void Dynamics::DynamicalEquations(
     const NumericalSolver::StateVector& x,
     NumericalSolver::StateVector& dxdt,
     const double& t,
-    const Array<Shared<Dynamics>>& aDynamicsArray,
-    const Instant& anInstant
+    const Array<Dynamics::Context>& aContextArray,
+    const Instant& anInstant,
+    const Shared<const Frame>& aFrame
 )
 {
-    for (Size i = 0; i < (Size)dxdt.size(); ++i)
+    dxdt.setZero();
+
+    const Instant nextInstant = anInstant + Duration::Seconds(t);
+
+    for (const Dynamics::Context& dynamicsInformation : aContextArray)
```

Comment:
Feels a bit more readable for me to keep the naming consistent

```suggestion
    for (const Dynamics::Context& dynamicsContext : aContextArray)
```